from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtWidgets import QFileDialog
from PyQt5.QtGui import QPixmap


class Ui_RunFaceAnalysis(object):
    def hideWindow(self, window):
        window.hide()

    def clicker(self):
        self.fname = QFileDialog.getOpenFileName(self, "Open File", "d:\\MyProjects\\FYP\\Main Project\\UnknownFaces",
                                                 "All Files (*);; PNG Files (*.png);;Jpg FIles (*.jpg)")

        self.pixmap = QPixmap(self.fname[0])
        self.imageView.setPixmap(self.pixmap)

    def setupUi(self, RunFaceAnalysis):
        RunFaceAnalysis.setObjectName("RunFaceAnalysis")
        RunFaceAnalysis.resize(593, 682)
        self.centralwidget = QtWidgets.QWidget(RunFaceAnalysis)
        self.centralwidget.setObjectName("centralwidget")
        self.openImage = QtWidgets.QPushButton(self.centralwidget, clicked = lambda : self.clicker())
        self.openImage.setGeometry(QtCore.QRect(80, 590, 201, 31))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(9)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.openImage.setFont(font)
        self.openImage.setObjectName("openImage")
        self.runAnalysis = QtWidgets.QPushButton(self.centralwidget)
        self.runAnalysis.setGeometry(QtCore.QRect(300, 590, 201, 31))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(9)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.runAnalysis.setFont(font)
        self.runAnalysis.setObjectName("runAnalysis")
        self.groupBox = QtWidgets.QGroupBox(self.centralwidget)
        self.groupBox.setGeometry(QtCore.QRect(10, 0, 571, 571))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(10)
        font.setBold(True)
        font.setWeight(75)
        self.groupBox.setFont(font)
        self.groupBox.setCheckable(False)
        self.groupBox.setObjectName("groupBox")
        self.horizontalLayoutWidget = QtWidgets.QWidget(self.groupBox)
        self.horizontalLayoutWidget.setGeometry(QtCore.QRect(100, 360, 391, 32))
        self.horizontalLayoutWidget.setObjectName("horizontalLayoutWidget")
        self.horizontalLayout = QtWidgets.QHBoxLayout(self.horizontalLayoutWidget)
        self.horizontalLayout.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.label = QtWidgets.QLabel(self.horizontalLayoutWidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.horizontalLayout.addWidget(self.label)
        self.detectedPerson = QtWidgets.QLabel(self.horizontalLayoutWidget)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.detectedPerson.setFont(font)
        self.detectedPerson.setText("")
        self.detectedPerson.setObjectName("detectedPerson")
        self.horizontalLayout.addWidget(self.detectedPerson)
        self.horizontalLayoutWidget_2 = QtWidgets.QWidget(self.groupBox)
        self.horizontalLayoutWidget_2.setGeometry(QtCore.QRect(100, 400, 391, 32))
        self.horizontalLayoutWidget_2.setObjectName("horizontalLayoutWidget_2")
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout(self.horizontalLayoutWidget_2)
        self.horizontalLayout_2.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        self.label_2 = QtWidgets.QLabel(self.horizontalLayoutWidget_2)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.horizontalLayout_2.addWidget(self.label_2)
        self.age = QtWidgets.QLabel(self.horizontalLayoutWidget_2)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.age.setFont(font)
        self.age.setText("")
        self.age.setObjectName("age")
        self.horizontalLayout_2.addWidget(self.age)
        self.horizontalLayoutWidget_3 = QtWidgets.QWidget(self.groupBox)
        self.horizontalLayoutWidget_3.setGeometry(QtCore.QRect(100, 440, 391, 32))
        self.horizontalLayoutWidget_3.setObjectName("horizontalLayoutWidget_3")
        self.horizontalLayout_3 = QtWidgets.QHBoxLayout(self.horizontalLayoutWidget_3)
        self.horizontalLayout_3.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
        self.label_3 = QtWidgets.QLabel(self.horizontalLayoutWidget_3)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.label_3.setFont(font)
        self.label_3.setObjectName("label_3")
        self.horizontalLayout_3.addWidget(self.label_3)
        self.gender = QtWidgets.QLabel(self.horizontalLayoutWidget_3)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.gender.setFont(font)
        self.gender.setText("")
        self.gender.setObjectName("gender")
        self.horizontalLayout_3.addWidget(self.gender)
        self.horizontalLayoutWidget_4 = QtWidgets.QWidget(self.groupBox)
        self.horizontalLayoutWidget_4.setGeometry(QtCore.QRect(100, 480, 391, 32))
        self.horizontalLayoutWidget_4.setObjectName("horizontalLayoutWidget_4")
        self.horizontalLayout_4 = QtWidgets.QHBoxLayout(self.horizontalLayoutWidget_4)
        self.horizontalLayout_4.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout_4.setObjectName("horizontalLayout_4")
        self.label_4 = QtWidgets.QLabel(self.horizontalLayoutWidget_4)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.horizontalLayout_4.addWidget(self.label_4)
        self.race = QtWidgets.QLabel(self.horizontalLayoutWidget_4)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.race.setFont(font)
        self.race.setText("")
        self.race.setObjectName("race")
        self.horizontalLayout_4.addWidget(self.race)
        self.horizontalLayoutWidget_5 = QtWidgets.QWidget(self.groupBox)
        self.horizontalLayoutWidget_5.setGeometry(QtCore.QRect(100, 520, 391, 32))
        self.horizontalLayoutWidget_5.setObjectName("horizontalLayoutWidget_5")
        self.horizontalLayout_5 = QtWidgets.QHBoxLayout(self.horizontalLayoutWidget_5)
        self.horizontalLayout_5.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout_5.setObjectName("horizontalLayout_5")
        self.label_5 = QtWidgets.QLabel(self.horizontalLayoutWidget_5)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.horizontalLayout_5.addWidget(self.label_5)
        self.emotion = QtWidgets.QLabel(self.horizontalLayoutWidget_5)
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(16)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.emotion.setFont(font)
        self.emotion.setText("")
        self.emotion.setObjectName("emotion")
        self.horizontalLayout_5.addWidget(self.emotion)
        self.imageView = QtWidgets.QLabel(self.groupBox)
        self.imageView.setGeometry(QtCore.QRect(30, 30, 521, 311))
        self.imageView.setText("")
        self.imageView.setAlignment(QtCore.Qt.AlignCenter)
        self.imageView.setObjectName("imageView")
        self.exit = QtWidgets.QPushButton(self.centralwidget, clicked = lambda :self.hideWindow(RunFaceAnalysis))
        self.exit.setGeometry(QtCore.QRect(80, 630, 421, 31))
        font = QtGui.QFont()
        font.setFamily("Segoe UI")
        font.setPointSize(9)
        font.setBold(True)
        font.setItalic(True)
        font.setWeight(75)
        self.exit.setFont(font)
        self.exit.setObjectName("exit")
        RunFaceAnalysis.setCentralWidget(self.centralwidget)
        self.statusbar = QtWidgets.QStatusBar(RunFaceAnalysis)
        self.statusbar.setObjectName("statusbar")
        RunFaceAnalysis.setStatusBar(self.statusbar)

        self.retranslateUi(RunFaceAnalysis)
        QtCore.QMetaObject.connectSlotsByName(RunFaceAnalysis)


    def retranslateUi(self, RunFaceAnalysis):
        _translate = QtCore.QCoreApplication.translate
        RunFaceAnalysis.setWindowTitle(_translate("RunFaceAnalysis", "MainWindow"))
        self.openImage.setText(_translate("RunFaceAnalysis", "Open Image"))
        self.runAnalysis.setText(_translate("RunFaceAnalysis", "Run Facial Analysis"))
        self.groupBox.setTitle(_translate("RunFaceAnalysis", "Unknown Faces Extraction and Analysis :"))
        self.label.setText(_translate("RunFaceAnalysis", "Detected Person: "))
        self.label_2.setText(_translate("RunFaceAnalysis", "Expected Age:"))
        self.label_3.setText(_translate("RunFaceAnalysis", "Gender: "))
        self.label_4.setText(_translate("RunFaceAnalysis", "Race: "))
        self.label_5.setText(_translate("RunFaceAnalysis", "Emotion/Expression: "))
        self.exit.setText(_translate("RunFaceAnalysis", "Exit"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    RunFaceAnalysis = QtWidgets.QMainWindow()
    ui = Ui_RunFaceAnalysis()
    ui.setupUi(RunFaceAnalysis)
    RunFaceAnalysis.show()
    sys.exit(app.exec_())
